server/models/User.js

```diff
@@ -48,10 +59,39 @@ export default class User {
     return Promise.resolve(this.strip());
   }
 
+  async login() {
+    try {
+      const authQuery = `SELECT users.id, authentication.password, authentication.username, users."fullName", users.email FROM authentication INNER JOIN users 
+      ON users."authId" = authentication.id 
+      WHERE authentication.username = '${this.username}'`;
+      const getUser = await client.query(authQuery);
+      const user = getUser.rows[0];
+      if (!user) return new Error('user not found');
+      const correctPassword = await bcrypt.compare(this.password, getUser.rows[0].password);
```

Comment:
Since this variable value will be a boolean, I think you can use a different name for this variable